src/Avalonia.Controls/Primitives/TextSelectionHandle.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.Media;
+
+namespace Avalonia.Controls.Primitives
+{
+    public class TextSelectionHandle : Thumb
+    {
+        internal SelectionHandleType SelectionHandleType { get; set; }
+
+        private Point _startPosition;
+
+        internal Point IndicatorPosition => IsDragging ? _startPosition.WithX(_startPosition.X) + _delta : Bounds.Position.WithX(Bounds.Position.X).WithY(Bounds.Y);
+
+        public bool IsDragging { get; private set; }
```

Comment:
It probably should be a direct property or private/internal. As otherwise developers would want to use it in styling.

Reply:
Made internal.